boot/freeldr/freeldr/ntldr/winldr.c

```diff
@@ -629,65 +689,24 @@ LoadWindowsCore(IN USHORT OperatingSystemVersion,
         SosEnabled = TRUE;
     }
 
-    if (OperatingSystemVersion > _WIN32_WINNT_NT4)
+    if (OperatingSystemVersion > _WIN32_WINNT_NT4 &&
+        NtLdrGetOption(BootOptions, "SAFEBOOT"))
     {
-        if (NtLdrGetOption(BootOptions, "SAFEBOOT"))
-        {
-            /* We found the SAFEBOOT option. */
-            FIXME("LoadWindowsCore: SAFEBOOT - TRUE (not implemented)\n");
-            SafeBoot = TRUE;
-        }
-        if (NtLdrGetOption(BootOptions, "PAE"))
-        {
-            /* We found the PAE option. */
-            FIXME("LoadWindowsCore: PAE - TRUE (not implemented)\n");
-            PaeEnabled = TRUE;
-        }
+        /* We found the SAFEBOOT option. */
+        FIXME("LoadWindowsCore: SAFEBOOT - TRUE (not implemented)\n");
+        SafeBoot = TRUE;
     }
 
-    if (OperatingSystemVersion > _WIN32_WINNT_WIN2K)
+    if (OperatingSystemVersion > _WIN32_WINNT_WIN2K &&
+        NtLdrGetOption(BootOptions, "BOOTLOGO"))
     {
-        if (NtLdrGetOption(BootOptions, "NOPAE"))
-        {
-            /* We found the NOPAE option. */
-            FIXME("LoadWindowsCore: NOPAE - TRUE (not implemented)\n");
-            PaeDisabled = TRUE;
-        }
-        if (NtLdrGetOption(BootOptions, "BOOTLOGO"))
-        {
-            /* We found the BOOTLOGO option. */
-            FIXME("LoadWindowsCore: BOOTLOGO - TRUE (not implemented)\n");
-            BootLogo = TRUE;
-        }
-
-        if (!LoaderBlock->SetupLdrBlock)
-        {
-            if (NtLdrGetOption(BootOptions, "EXECUTE"))
-            {
-                /* We found the EXECUTE option. */
-                FIXME("LoadWindowsCore: EXECUTE - TRUE (not implemented)\n");
-                NoexecuteDisabled = TRUE;
-            }
-            if (NtLdrGetOption(BootOptions, "NOEXECUTE=ALWAYSOFF"))
-            {
-                /* We found the NOEXECUTE=ALWAYSOFF option. */
-                FIXME("LoadWindowsCore: NOEXECUTE=ALWAYSOFF - TRUE (not implemented)\n");
-                NoexecuteDisabled = TRUE;
-            }
-            if (NtLdrGetOption(BootOptions, "NOEXECUTE"))
-            {
-                /* We found the NOEXECUTE option. */
-                FIXME("LoadWindowsCore: NOEXECUTE - TRUE (not implemented)\n");
-                NoexecuteEnabled = TRUE;
-            }
-        }
+        /* We found the BOOTLOGO option. */
+        FIXME("LoadWindowsCore: BOOTLOGO - TRUE (not implemented)\n");
+        BootLogo = TRUE;
     }
 
-    if (SafeBoot)
-    {
-        PaeDisabled = TRUE;
-        NoexecuteDisabled = TRUE;
-    }
+    PaeModeOn = WinLdrIsPaeSupported(LoaderBlock, BootOptions, OperatingSystemVersion, KernelFileName, HalFileName);
+    FIXME("WinLdrIsPaeSupported: PaeModeOn %X\n", PaeModeOn);
```

Comment:
I don't see the value in this `FIXME`, we have enough debug output around here. Please remove